### ISSUE TITLE
Regex failing with unmatched parentheses

### ISSUE THREAD

**Issue by ghost** [oven-sh/bun#2926]
### What version of Bun is running?

0.6.1

### What platform is your computer?

Darwin 22.4.0 arm64 arm

### What steps can reproduce the bug?

This snippet of code from sql-highlight was working on bun v0.5.8 but broke in bun v0.6.1
```js
const keywords = ["ADD", "CREATE"];
const SPLIT_CHARS = '[^a-zA-Z_]'
new RegExp(`(^|${SPLIT_CHARS})(${keywords.join('|')})(?=${SPLIT_CHARS}|$)`, 'gi')
```

### What is the expected behavior?

It should not throw an error

### What do you see instead?
```js
4 | new RegExp(`(^|${SPLIT_CHARS})(${keywords.join('|')})(?=${SPLIT_CHARS}|$)`, 'gi')
   ^
SyntaxError: Invalid regular expression: unmatched parentheses
```

### Additional information

It works fine in nodejs and the browser.

**Comment by pepper-lees** [comment]
I believe this issue is caused by Bun messing up the template strings. Try changing the code to this:

```js
const keywords = ['ADD', 'CREATE'];
const SPLIT_CHARS = '[^a-zA-Z_]';
let str = `(^|${SPLIT_CHARS})(${keywords.join('|')})(?=${SPLIT_CHARS}|$)`;
console.log(str);
```

then run with node and bun. See that Bun actually messes up the template string and causes the parentheses to be mismatched:


```
$ bun test.js
(^|[^a-zA-Z_]|$)ADD|CREATE)(?=[^a-zA-Z_]|$)
$ node test.js
(^|[^a-zA-Z_])(ADD|CREATE)(?=[^a-zA-Z_]|$)
```

I have another issue that I'm going to write out which throws the same SyntaxError, but I don't believe is caused by the same template string issue.

**Comment by Jarred-Sumner** [comment]
Can you try on the canary build of Bun? 

<img width="354" alt="image" src="https://github.com/oven-sh/bun/assets/709451/6434913d-af24-4963-aa62-686163ab452f">

**Comment by Jarred-Sumner** [comment]
Fixed in Bun v0.6.2

**Comment by ghost** [comment]
Thanks it works now in v0.6.2.
